### ISSUE TITLE
401 response from Recurly throws vague exception

### ISSUE THREAD

**Issue by pavellishin** [killbilling/recurly-java-library#123]
When receiving a 401 response, Recurly doesn't send any XML data, but a simple string: 

`HTTP Basic: Access denied.`

This causes the following exception:

```
Exception in thread "main" com.ning.billing.recurly.RecurlyAPIException: Unspecified API Error
	at com.ning.billing.recurly.RecurlyClient.callRecurly(RecurlyClient.java:1004)
	at com.ning.billing.recurly.RecurlyClient.callRecurlySafe(RecurlyClient.java:941)
	at com.ning.billing.recurly.RecurlyClient.doGETWithFullURL(RecurlyClient.java:895)
	at com.ning.billing.recurly.RecurlyClient.doGET(RecurlyClient.java:888)
	at com.ning.billing.recurly.RecurlyClient.getPlan(RecurlyClient.java:629)
	at com.bazaarvoice.connections.greenlight.api.Recurly.main(Recurly.java:19)
13:38:41.479 [main] DEBUG c.ning.billing.recurly.RecurlyClient - Unable to extract error
java.io.IOException: javax.xml.stream.XMLStreamException: ParseError at [row,col]:[1,1]
Message: Content is not allowed in prolog.
	at com.fasterxml.jackson.dataformat.xml.util.StaxUtil.throwXmlAsIOException(StaxUtil.java:24)
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._createParser(XmlFactory.java:557)
	at com.fasterxml.jackson.dataformat.xml.XmlFactory.createParser(XmlFactory.java:429)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2779)
	at com.ning.billing.recurly.RecurlyClient.callRecurly(RecurlyClient.java:1000)
	at com.ning.billing.recurly.RecurlyClient.callRecurlySafe(RecurlyClient.java:941)
	at com.ning.billing.recurly.RecurlyClient.doGETWithFullURL(RecurlyClient.java:895)
	at com.ning.billing.recurly.RecurlyClient.doGET(RecurlyClient.java:888)
	at com.ning.billing.recurly.RecurlyClient.getPlan(RecurlyClient.java:629)
	at com.bazaarvoice.connections.greenlight.api.Recurly.main(Recurly.java:19)
Caused by: javax.xml.stream.XMLStreamException: ParseError at [row,col]:[1,1]
Message: Content is not allowed in prolog.
	at com.sun.org.apache.xerces.internal.impl.XMLStreamReaderImpl.next(XMLStreamReaderImpl.java:601)
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._initializeXmlReader(XmlFactory.java:648)
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._createParser(XmlFactory.java:555)
	... 8 common frames omitted
```

Only with logging set to WARN do you get a clue as to what is going on:

```
13:37:33.996 [main] WARN  c.ning.billing.recurly.RecurlyClient - Recurly error whilst calling: https://bazaarvoice.recurly.com:443/v2/plans/alpine-test-1?per_page=20
HTTP Basic: Access denied.
```

**Comment by pavellishin** [comment]
Probably related to #116

**Comment by bhelx** [comment]
I agree, it's too vague. It should fail hard and let you know exactly what is wrong. Any feedback you could give me on an implementation would be helpful. This works but may require a little discussion https://github.com/killbilling/recurly-java-library/pull/151

**Comment by bhelx** [comment]
Closed with #151
